src/Layer/LabelLayer.js

```diff
@@ -57,6 +57,49 @@ class LabelLayer extends Layer {
         this.buildExtent = true;
 
         this.labelDomelement = config.domElement;
+
+        if (
+            this.style
+            && (this.style.point && (this.style.point.color || this.style.point.line))
+            && !(this.style.icon && (this.style.icon.source || this.style.icon.key))
+        ) {
+            // Create a canvas
+            const canvas = document.createElement('canvas');
+
+            // Set canvas dimensions
+            const width = this.style.point.radius + this.style.point.width;
+            canvas.width = 2 * width;
+            canvas.height = 2 * width;
+
+            if (canvas.getContext) {
+                const ctx = canvas.getContext('2d');
+
+                // Draw a circle with dimensions specified in layer style
+                ctx.beginPath();
+                ctx.arc(width, width, this.style.point.radius, 0, 2 * Math.PI);
+
+                // Apply transparency to the circle if specified in layer style
+                ctx.globalAlpha = this.style.point.opacity;
+
+                // Fill the circle if specified in layer style
+                if (this.style.point.color) {
+                    ctx.fillStyle = this.style.point.color;
+                    ctx.fill();
+                }
+
+                // Draw the circle's edges if specified in layer style
+                if (this.style.point.line) {
+                    ctx.lineWidth = this.style.point.width;
+                    ctx.strokeStyle = this.style.point.line;
+                    ctx.stroke();
+                }
+
+                this.style.icon = this.style.icon || {};
+                this.style.icon.source = canvas.toDataURL();
+                this.style.point.color = undefined;
+                this.style.point.line = undefined;
```

Comment:
I'm not sure to keep this part....

Reply:
Changing `this.style.icon.source` is necessary so that the canvas drawing is displayed as icon. The following two lines prevent drawing points projected on ground, as can be seen in the attached screenshot. Moreover, they do not actually change the `Style` object defined by the user. Do you still think I should remove them ?
![Capture d’écran du 2022-02-15 12-00-18](https://user-images.githubusercontent.com/73115044/154049177-68814260-a056-410f-9ac8-2c50ad89ee42.png)